File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Building.java
package io.github.dot166.jlib.game;

import static io.github.dot166.jlib.game.GameView.PARAMS;
import static io.github.dot166.jlib.game.GameView.irand;

import android.content.Context;

public class Building extends Scenery {
    public Building(Context context) {
        super(context);

        w = irand(PARAMS.BUILDING_WIDTH_MIN, PARAMS.BUILDING_WIDTH_MAX);
        h = 0; // will be setup later, along with z

        setTranslationZ(PARAMS.SCENERY_Z);
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Building.kt
package io.github.dot166.jlib.game

import android.content.Context

class Building(context: Context?) : Scenery(context) {
    init {
        w = GameView.irand(GameView.PARAMS.BUILDING_WIDTH_MIN, GameView.PARAMS.BUILDING_WIDTH_MAX)
        h = 0 // will be setup later, along with z

        translationZ = GameView.PARAMS.SCENERY_Z
    }
}